python/ray/workflow/http_event_provider.py

```diff
@@ -0,0 +1,157 @@
+import asyncio
+from typing import Dict, Any
+from fastapi import FastAPI, Request, HTTPException
+
+import ray
+from ray import serve
+from ray.workflow import workflow_context
+from ray.workflow.event_listener import EventListener
+from ray.workflow.common import Event
+
+
+class WorkflowEventHandleError(Exception):
+    def __init__(self, workflow_id: str, what_happen: str):
+        self.message = f"Workflow[id={workflow_id}] HTTP event handle failed: {what_happen}"
+        super().__init__(self.message)
+
+
+app = FastAPI()
+
+
+@serve.deployment(name="myEventProvider", route_prefix="/event")
+@serve.ingress(app)
+class HTTPEventProvider():
+    """HTTPEventProvider is defined to be a Ray Serve deployment with route_prefix='/event',
+    which will receive external events via an HTTP endpoint. It supports FastAPI, e.g. post.
+    It responds to both poll_for_event() and event_checkpointed() from an HTTPListener instance.
+
+    HTTPListener class is designed to work with the current workflow.wait_for_event()
+    implementation, where an HTTPListener instance will be initiated by the get_message() and
+    message_committed() of the workflow.wait_for_event().
+
+    Example definition
+    ==================
+
+    ```
+    class HTTPEventProvider:
+
+        def __init__(self):
+
+        @app.post("/send_event/{workflow_id}")
+        async def create_event_payload(self, workflow_id: str, req: Request):
+
+        async def get_event_payload(self, workflow_id, event_key):
+
+        async def report_checkpointed(self, event, confirmation):
+
+    ```
+
+    Example Usage
+    =============
+    >>> from ray.workflow.http_event_provider import HTTPEventProvider, HTTPListener
+    >>> ray.init(address='auto', namespace='serve')
+    >>> serve.start(detached=True)
+    >>> HTTPEventProvider.deploy()
+    >>> event_step = workflow.wait_for_event( # doctest: +SKIP
```

Comment:
avoid step. we don't have step concept anymore.

Reply:
Done